openjscad.ts TS2394 overload signature

Well, there is still a compiling error using 2.4 typescript version:

src/core/openjscad.ts(133,21): error TS2394: Overload signature is not compatible with function implementation.

It's something that has been added between typescript 2.4.0 and typescript 2.4.1 AFAIK, makerjs compiles with 2.4.0 but not 2.4.1

--- a/package.json
+++ b/package.json
@@ -113,12 +113,10 @@
   "dependencies": {
     "@types/bezier-js": "^0.0.6",
     "@types/node": "^7.0.5",
-    "@types/openjscad": "^0.0.28",
     "@types/opentype.js": "^0.0.0",
     "@types/pdfkit": "^0.7.34",
     "bezier-js": "^2.1.0",
     "clone": "^1.0.2",
-    "graham_scan": "^1.0.4",
-    "openjscad-csg": "^1.0.3"
+    "graham_scan": "^1.0.4"
   }
 }

--- a/src/core/openjscad.ts
+++ b/src/core/openjscad.ts
@@ -32,10 +32,10 @@
         if (!facetSize) return;
 
         var circle = new paths.Circle([0, 0], arcOrCircle.radius);
-        
+
         var length = measure.pathLength(circle);
         if (!length) return;
-        
+
         return length / facetSize;
     }
 
@@ -112,7 +112,7 @@
 
         for (var pathId in modelContext.paths) {
             var pathContext = modelContext.paths[pathId];
-            
+
             var fn = first ? beginMap[pathContext.type] : appendMap[pathContext.type];
 
             if (fn) {
@@ -263,7 +263,7 @@
 
         var f = new Function('CAG', 'CSG', script);
 
-        var csg = <CSG>f(container.CAG, container.CSG);
+        var csg = f(container.CAG, container.CSG) as CSG;
 
         return csg.toStlString();
     }
@@ -298,4 +298,41 @@
         [modelId: string]: IOpenJsCadOptions;
     }
 }
- 
+
+//minimial OpenJscad - CSG declaration
+declare class CxG {
+    toStlString(): string;
+}
+
+declare class CSG extends CxG {
+}
+
+declare class Vector2D extends CxG {
+}
+
+declare namespace CSG {
+    interface IRadiusOptions {
+        radius?: number;
+        resolution?: number;
+    }
+    interface ICircleOptions extends IRadiusOptions {
+        center?: Vector2D | number[];
+    }
+    interface IArcOptions extends ICircleOptions {
+        startangle?: number;
+        endangle?: number;
+        maketangent?: boolean;
+    }
+    interface IEllpiticalArcOptions extends IRadiusOptions {
+        clockwise?: boolean;
+        large?: boolean;
+        xaxisrotation?: number;
+        xradius?: number;
+        yradius?: number;
+    }
+}
+declare interface CAG_extrude_options {
+    offset?: number[];
+    twistangle?: number;
+    twiststeps?: number;
+}

